docs/Streams.md

```diff
@@ -186,3 +193,19 @@ MsQuic does not enforce it, and it is legal for an application to provide less b
 
 After the initial receive window is full, flow control will ensure that the peer does not send more data than there is buffer space available.
 However, the application should still provide enough buffer space to keep flow control from impacting performances.
+
+## Receive Shutdown
+
+The receiver can abortively shutdown a stream receive direction by calling [`StreamShutdown`](api/StreamShutdown.md) 
+with the `QUIC_STREAM_SHUTDOWN_FLAG_ABORT_RECEIVE` option.
+
+# Closing a Stream
+
+Once all the directions relevant for a stream have been shutdown, the application receives a
+`QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE` event.
+
+The application must then close the stream using [`StreamClose`](api/StreamClose.md), and can release its notification
+handling context.
+
+Closing a stream before the `QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE` is only allowed if the stream was not started yet.
+Otherwise, it will result in the connection being closed abortively.
```

Comment:
This is not strictly correct. If you close a stream before you've shut down the stream, the stream will be aborted implicitly (with error code 0). This is not a great behavior because we don't actually know what error 0 will mean - so generally an app really shouldn't do that.

But if the application aborts a stream with an error code, it may immediately close the stream if doesn't care to wait for the completion. This is a completely reasonable behavior.

Reply:
Hum, I think I wrote this because of this comment in our code, in `QuicStreamClose`:

```
            //
            // TODO - If the stream hasn't been aborted already, then this is a
            // fatal error for the connection. The QUIC transport cannot "just
            // pick an error" to shutdown the stream with. It must abort the
            // entire connection.
            //
```

Is it no longer the case? I will delete the comment if so.

Reply:
Well, we don't actually kill the connection though; just log a warning. My description above still holds I think.